projectq/backends/_sim/_cppkernels/simulator.hpp

```diff
@@ -209,46 +222,85 @@ class Simulator{
     }
 
     template <class F, class QuReg>
-    void emulate_math(F const& f, QuReg quregs, std::vector<unsigned> ctrl,
-                      unsigned num_threads=1){
+    void emulate_math(F const& f, QuReg quregs, const std::vector<unsigned>& ctrl,
+                      bool parallelize = false){
```

Comment:
Is there a particular reason why the C++ code always call this function with `parallelize` set to `true` and the Python code `false`?

Reply:
Yes!
Python has a GIL (global interpreter lock) so there is no benefit of running python code from multiple threads... they'll block each other all the time which probably creates more overhead.

Reply:
True, I forgot that ;-)